app/controllers/katello/concerns/api/v2/hosts_controller_extensions.rb

```diff
@@ -36,6 +36,9 @@ def host_collections
 
         def set_content_view_environments
           content_facet_attributes = params.dig(:host, :content_facet_attributes)
+          if @host.present? && @host&.content_facet.blank?
+            @host.content_facet = Katello::Host::ContentFacet.create!(host: @host)
+          end
           return if content_facet_attributes.blank? || @host&.content_facet.blank? ||
             (cve_params[:content_view_id].present? && cve_params[:lifecycle_environment_id].present?)
```

Comment:
won't that now create a content facet for *every* host, even those that don't get a CVE?
maybe:
```suggestion
          return if content_facet_attributes.blank? ||
            (cve_params[:content_view_id].present? && cve_params[:lifecycle_environment_id].present?)
          if @host.present? && @host&.content_facet.blank?
            @host.content_facet = Katello::Host::ContentFacet.create!(host: @host)
          end
```

Reply:
ugh, that diff renders ugly. I was suggesting: move the `create!` after the `return`, but drop the `@host&.content_facet.blank? ` part of the `return` as that'd be always `false` now anyway.